judgels-backends/uriel/uriel-app/src/main/java/judgels/uriel/contest/scoreboard/icpc/IcpcScoreboardProcessor.java

```diff
@@ -22,6 +35,133 @@ public Scoreboard parseFromString(ObjectMapper mapper, String json) {
+            problemJids.forEach(p -> penaltyMap.put(p, 0L));
+            Map<String, IcpcScoreboardProblemState> stateMap = new HashMap<>();
+            problemJids.forEach(p -> stateMap.put(p, IcpcScoreboardProblemState.NOT_ACCEPTED));
+
+            for (Submission submission : submissions) {
+                if (!contestantJids.contains(submission.getUserJid())) {
+                    continue;
+                }
+
+                if (!problemJidsSet.contains(submission.getProblemJid())) {
+                    continue;
+                }
+
+                if (!stateMap.get(submission.getProblemJid()).equals(IcpcScoreboardProblemState.NOT_ACCEPTED)) {
+                    continue;
+                }
+
+                if (!submission.getLatestGrading().isPresent()
+                        || submission.getLatestGrading().get().getVerdict().equals(Submission.PENDING)) {
+                    continue;
+                }
+
+                attemptsMap.put(submission.getProblemJid(), attemptsMap.get(submission.getProblemJid()) + 1);
+
+                long submissionPenalty = submissionPenaltyInMillisecond(
+                        submission.getTime(),
+                        contestantStartTimesMap.get(contestantJid),
+                        contest.getBeginTime());
+                penaltyMap.put(submission.getProblemJid(), convertPenaltyToMinutes(submissionPenalty));
+
+                if (submission.getLatestGrading().get().getVerdict().equals(Submission.ACCEPTED)) {
+                    if (firstSolveId.get(submission.getProblemJid()).equals(submission.getJid())) {
+                        stateMap.put(
+                                submission.getProblemJid(),
+                                IcpcScoreboardProblemState.FIRST_ACCEPTED);
+                    } else {
+                        stateMap.put(
+                                submission.getProblemJid(),
+                                IcpcScoreboardProblemState.ACCEPTED);
+                    }
+
+                    lastAcceptedPenalty = submissionPenalty;
+                }
+            }
+
+            scoreboardEntryList.add(new IcpcScoreboardEntry.Builder()
+                    .rank(0)
+                    .contestantJid(contestantJid)
+                    .totalAccepted((int) stateMap.values()
+                            .stream()
+                            .filter(s -> s != IcpcScoreboardProblemState.NOT_ACCEPTED)
+                            .count())
+                    .totalPenalties(penaltyMap.keySet().stream()
+                            .filter(p -> stateMap.get(p) != IcpcScoreboardProblemState.NOT_ACCEPTED)
+                            .mapToInt(p -> ((attemptsMap.get(p) - 1)
+                                    * (int) icpcStyleModuleConfig.getWrongSubmissionPenalty())
+                                    + penaltyMap.get(p).intValue())
+                            .sum())
+                    .lastAcceptedPenalty((int) lastAcceptedPenalty)
```

Comment:
I think we must fix the type (it should be long, not int) (also for totalPenalties)

Reply:
Okay (y)